src/test/store/receive-profile.test.js

```diff
@@ -901,6 +903,147 @@ describe('actions/receive-profile', function() {
       ).toMatchSnapshot();
     });
   });
+
+  describe('retrieveProfilesToCompare', function() {
+    function fetch200Response(profile: Profile) {
+      return {
+        ok: true,
+        status: 200,
+        headers: {
+          get: () => 'appliciation/json',
+        },
+        json: () => Promise.resolve(profile),
+      };
+    }
+
+    // This turns a processed profile into a processed profile that would come
+    // from the store. Especially the stringTable needs to be serialized to a
+    // stringArray.
+    function unhydrateProfile(profile: Profile): Object {
+      return {
+        ...profile,
+        threads: profile.threads.map(thread => {
+          const { stringTable, ...restThread } = thread;
+          return {
+            ...restThread,
+            stringArray: stringTable.serializeToArray(),
+          };
+        }),
+      };
+    }
+
+    async function setup(urlSearch1: string, urlSearch2: string): * {
+      const { profile: profile1 } = getProfileFromTextSamples(
+        `A  B  C  D  E`,
+        `G  H  I  J  K`
+      );
+      const { profile: profile2 } = getProfileFromTextSamples(
+        `L  M  N  O  P`,
+        `Q  R  S  T  U`
+      );
+
+      profile1.threads.forEach(thread =>
+        addMarkersToThreadWithCorrespondingSamples(thread, [
+          ['A', 1, { startTime: 1, endTime: 3 }],
+          ['B', 2, null],
+          ['C', 3, null],
+          ['D', 4, null],
+          ['E', 5, null],
+        ])
+      );
+      profile2.threads.forEach(thread =>
+        addMarkersToThreadWithCorrespondingSamples(thread, [
+          ['F', 1, { startTime: 1, endTime: 3 }],
+          ['G', 2, null],
+          ['H', 3, null],
+          ['I', 4, null],
+          ['J', 5, null],
+        ])
+      );
+
+      window.fetch
+        .mockResolvedValueOnce(fetch200Response(unhydrateProfile(profile1)))
+        .mockResolvedValueOnce(fetch200Response(unhydrateProfile(profile2)));
+
+      const fakeUrl1 = `https://fakeurl.com/public/fakehash1/?${urlSearch1}&v=3`;
+      const fakeUrl2 = `https://fakeurl.com/public/fakehash2/?${urlSearch2}&v=3`;
+
+      const { dispatch, getState } = blankStore();
+      await dispatch(retrieveProfilesToCompare([fakeUrl1, fakeUrl2]));
+
+      const resultProfile = ProfileViewSelectors.getProfile(getState());
+      return { profile1, profile2, dispatch, getState, resultProfile };
+    }
+
+    beforeEach(function() {
+      // The stub makes it easy to return different values for different
+      // arguments. Here we define the default return value because there is no
+      // argument specified.
+      window.fetch = jest.fn();
+      window.fetch.mockImplementation(() =>
+        Promise.reject(new Error('No more answers have been configured.'))
+      );
+    });
+
+    afterEach(function() {
+      delete window.fetch;
```

Comment:
Doesn't our automock config do this for us?

Reply:
because `window.fetch` doesn't exist in node, we can't use `jest.spyOn` on it :( this works only on existing properties.